### ISSUE TITLE
Templating engine breaks SVG component scaled rendering by lowercasing viewBox attribute

### ISSUE THREAD

**Issue by JeremiahSanders** [fsbolero/Bolero#256]
**TL;DR**: SVGs use [a `viewBox` attribute](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox) which is _case-sensitive_. This attribute is essential for rendering resizable images. The Bolero `Template<>` implementation converts `viewBox` attributes to `viewbox` (all lowercase), which breaks scaled rendering.

> **This is _not_ a newly-introduced problem in `0.20.12`.** I've been using Bolero to develop an [SVG-heavy realtime WASM game (Roadkill Ruffians)](https://roadkillruffians.app/) since Bolero `0.10.1-preview9`. The problem has existed since roughly that point, but I worked around it by creating helper functions utilizing the `list`-based rendering to manually specify SVG `attr` names.
>
> Example: `let viewBox (value: obj): Attr = Attr("viewBox", box value)` was the function I used extensively to create the functional SVGs.
>
> However, the **breaking** changes introduced in `0.20` have required an extensive component building refactor. I can no longer just build `Attr list` and pass it to `Html.svg`. Thus, the inability to use SVG templates (due to the attribute being coerced to lowercase) is causing increased difficulty.

## Sample SVG

Below is a simple, hollow circle which renders to roughly fill a `200x200` area.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<svg width="200" height="200" version="1.1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
    <circle cx="50" cy="50" r="45" fill="none" stroke="#000" stroke-width="3"/>
</svg>
```

> Note the case-sensitive `viewBox` attribute.

## Sample SVG Template

Below is the same SVG, but including Bolero Holes to specify the circle fill and stroke at runtime.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<svg width="200" height="200" version="1.1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
    <circle cx="50" cy="50" r="45" fill="${CircleFill}" stroke="${CircleStroke}" stroke-width="3"/>
</svg>
```

> Note the case-sensitive `viewBox` attribute.

### Rendered SVG Template

The template above, rendered by Bolero `0.20.12`, generates the following (copied using "copy element" in Edge developer tools):

```xml
<svg width="200" height="200" version="1.1" viewbox="0 0 100 100" xmlns="http://www.w3.org/2000/svg"><!--!-->
    <circle cx="50" cy="50" r="45" fill="none" stroke="#000" stroke-width="3"></circle><!--!-->
</svg>
```

> Note the conversion of `viewBox` to `viewbox` in the version rendered by Bolero.

## Working Example

A minimal example project exhibiting this behavio, based on the latest NuGet project template, can be seen at: [https://github.com/JeremiahSanders/fsbolero-bolero-svg-viewbox](https://github.com/JeremiahSanders/fsbolero-bolero-svg-viewbox).

The commit which modifies the template with the minimum changes to show this issue: [https://github.com/JeremiahSanders/fsbolero-bolero-svg-viewbox/tree/f9281c33b8793dc0fb5a282659e975767831d738](https://github.com/JeremiahSanders/fsbolero-bolero-svg-viewbox/tree/f9281c33b8793dc0fb5a282659e975767831d738)

**Comment by JeremiahSanders** [comment]
After examining [`Bolero.Templating.Parsing.GetDoc`](https://github.com/fsbolero/Bolero/blob/fbf5ffef49c732347dd79b69ca0e8a7ce38a6554/src/Bolero.Templating.Provider/Parsing.fs#L373), it appears that the problem is caused by `HtmlAgilityPack`.
A matching symptom is addressed in [this StackOverflow answer](https://stackoverflow.com/a/50352029).

Based upon the above, it appears that SVG templates (and any others which need case-sensitive output) might render successfully by changing the properties applied to [this](https://github.com/fsbolero/Bolero/blob/fbf5ffef49c732347dd79b69ca0e8a7ce38a6554/src/Bolero.Templating.Provider/Parsing.fs#L374) object in [`Bolero.Templating.Parsing.GetDoc`](https://github.com/fsbolero/Bolero/blob/fbf5ffef49c732347dd79b69ca0e8a7ce38a6554/src/Bolero.Templating.Provider/Parsing.fs#L373). Specifically, setting its `OptionOutputOriginalCase` to `true`.

The following script, when executed in [.NET Fiddle](https://dotnetfiddle.net/), shows proper rendering of `viewBox` attribute, due to use of `OptionOutputOriginalCase`:

```f#
open HtmlAgilityPack

let svg = """<svg width="200" height="200" version="1.1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg">
    <circle cx="50" cy="50" r="45" fill="${CircleFill}" stroke="${CircleStroke}" stroke-width="3"/>
</svg>"""

let svgDoc = new HtmlDocument()

svgDoc.OptionOutputOriginalCase <- true
svgDoc.LoadHtml(svg)

let svgBody = svgDoc.DocumentNode.SelectSingleNode("//svg")

printfn "%O" (svgBody.OuterHtml)
```

**Comment by Tarmil** [comment]
Fixed in v0.20.17.
